core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/time/TimeFormat.java

```diff
@@ -14,31 +14,45 @@
  */
 public class TimeFormat implements DataInstruction {
 
-    private final String ORIGINAL_TIME_REPRESENTATION_KEY = "OriginalDateRepresentation";
+    private final static String ORIGINAL_TIME_REPRESENTATION = "OriginalTimeRepresentation";
 
-    private final String DESIRED_TIME_REPRESENTATION_KEY = "DesiredDateRepresentation";
+    private final static String ORIGINAL_TIME_REPRESENTATION_FORMAT = "OriginalTimeRepresentationFormat";
 
-    private final Pattern INPUT_PARAMETER_PATTERN = Pattern.compile(
-            "\\s*\"?(?<" + ORIGINAL_TIME_REPRESENTATION_KEY + ">\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}.\\d{3})\"?\\s*,\\s*\"?(?<"
-                    + DESIRED_TIME_REPRESENTATION_KEY + ">[^\"]+)\"?\\s*");
+    private final static String DESIRED_TIME_REPRESENTATION = "DesiredTimeRepresentation";
 
-    private final SimpleDateFormat ORIGINAL_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+    private final static Pattern INPUT_PARAMETERS_PATTERN = Pattern.compile(
+            "\\s*\"?(?<" + ORIGINAL_TIME_REPRESENTATION + ">[^\"]+)\"?\\s*" +
+                    "(,\\s*\"?(?<" + ORIGINAL_TIME_REPRESENTATION_FORMAT + ">[^\"]+)\"?\\s*)?"+
+                    ",\\s*\"?(?<" +  DESIRED_TIME_REPRESENTATION + ">[^\"]+)\"?");
+
```

Comment:
Maybe its more clear to create 2 input patterns like we did for text.substring?

Reply:
as you whish, I can change it if you want